src/lib/FlightTasks/tasks/AutoLineSmoothVel/FlightTaskAutoLineSmoothVel.cpp

```diff
@@ -247,10 +247,17 @@ void FlightTaskAutoLineSmoothVel::_prepareSetpoints()
 			Vector2f pos_traj_to_dest(pos_sp_xy - pos_traj);
 			Vector2f prev_to_pos(pos_traj - Vector2f(_prev_wp));
 			Vector2f u_pos_traj_to_dest_xy(Vector2f(pos_traj_to_dest).unit_or_zero());
+			const bool has_reached_altitude = fabsf(_position_setpoint(2) - _position(2)) < _param_nav_mc_alt_rad.get();
```

Comment:
We should compare the position of the trajectory and not the one of the drone here. I'll merge and make the modification in my branch.